src/aggregators/rssa.jl

```diff
@@ -98,35 +98,43 @@ function execute_jumps!(p::RSSAJumpAggregation, integrator, u, params, t)
 end
 
 # calculate the next jump / jump time
-@fastmath function generate_jumps!(p::RSSAJumpAggregation, integrator, u, params, t)
+function generate_jumps!(p::RSSAJumpAggregation, integrator, u, params, t)
     sum_rate = p.sum_rate
     # if no more events possible there is nothing to do
     if nomorejumps!(p, sum_rate)
         return nothing
     end
     # next jump type
     @unpack ma_jumps, rates, cur_rate_high, cur_rate_low, rng = p
-    #rerl        = one(sum_rate)
+    num_majumps = get_num_majumps(ma_jumps)
     rerl        = zero(sum_rate)
 
-    r      = rand(rng) * sum_rate
-    jidx   = linear_search(cur_rate_high, r)
-    rerl  += randexp(rng)
-    @inbounds while rejectrx(ma_jumps, rates, cur_rate_high, cur_rate_low, rng, u, jidx, params, t)
+    r    = rand(rng) * sum_rate
+    jidx = linear_search(cur_rate_high, r)
+    if iszero(jidx)
```

Comment:
We might want to update this to recalculate `sum_rate` and if it is zero return as it currently does, and if it is not zero redo the search with an updated `r`. This could mitigate some potential floating point errors when the running sum is larger than the actual total propensity (which could also lead to getting a zero index). I'm going to leave that for a next PR though since I want to think more about how to handle this.